shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/DynamicFeatureManager.java

```diff
@@ -0,0 +1,128 @@
+ * gen_snapshot during compilation. The loading unit id is passed down through the engine and invokes
+ * donwloadDynamicFeature. Once the feature module is downloaded, loadAssets and loadDartLibrary are
+ * invoked which discovers, unpacks, and opens any assets or shared libs included in the dynamic
+ * feature module and passes the results back to the engine.
+ *
+ * <p>The methods here may also be called manually via platform channel messages.
+ *
+ * <p>A dynamic feature module is uniquely identified by a module name. Each feature module may
+ * contain one or more loading units, uniquely identified by the loading unit ID.
+ */
+public interface DynamicFeatureManager {
+  /**
+   * Request that the feature module be downloaded and installed.
+   *
+   * <p>This method is called by Flutter when loadLibrary() is called on a dart deferred library
+   * which relies on this method to download the requested module under the hood. This method may
+   * also be manually invoked via a system channel message.
+   *
+   * <p>This method begins the download and installation of the specified feature module. For example,
+   * the Play Store dynamic delivery implementation uses SplitInstallManager to request the download of
+   * the module. Download is not complete when this method returns. Upon completion of
+   * download, loadAssets and loadDartLibrary should be called to complete the dynamic feature load
+   * process.
+   *
+   * <p>Both parameters are not always necessary to identify which module to install. Asset-only modules
+   * do not have an associated loadingUnitId. Instead, an invalid ID like -1 may be passed to download
+   * only with moduleName. On the other hand, it can be possible to resolve the moduleName based
+   * on the loadingUnitId. This resolution is done if moduleName is null. At least one of
+   * loadingUnitId or moduleName must be valid or non-null.
+   *
+   * @param loadingUnitId The unique identifier associated with a dart deferred library. This id is
+   *                      assigned by gen_snapshot and can be referenced via bundle_config.yaml.
+   *                      This ID is primarily used in loadDartLibrary to indicate to dart which
+   *                      dart library is being loaded. Loading unit ids range from 0 to the number
+   *                      existing loading units. Passing a negative loading unit id indicates
+   *                      that no dart deferred library should be loaded after download completes.
+   *                      Passing a loadingUnitId larger than the highest valid loading unit's id will
+   *                      cause the dart loadLibrary() to complete with a failure.
+   *
+   * @param moduleName    The dynamic feature module name as defined in bundle_config.yaml. Flutter's
+   *                      default dynamic feature system stores a mapping of loadingUnitId -> moduleName
```

Comment:
Not sure what the subclasser should do with this sentence. What are we instructing the implementer to do?